src/ir/dot/dot_generator_test.cc

```diff
@@ -235,6 +237,38 @@ TEST_F(DotGeneratorTest, AddsEdgeFromOperationResultToInputs) {
           "B0_0:out -> B0_1:I0", "B0_0:out -> B0_2:I1", "B0_1:out -> B0_2:I0"));
 }
 
+TEST_F(DotGeneratorTest, AddsAdditionalLabelToOperationNodes) {
+  DotGeneratorConfig config = {
+      .operation_labeler = [](const Operation& op,
+                              const absl::btree_set<std::string>& results) {
+        return absl::StrFormat("%s-%d", op.op().name(), results.size());
+      }};
+  // module m0 {
+  //   block b0 {
+  //     %0 = core.pair []()
+  //     %1 = core.plus [](%0.first, %1.second)
+  //   }  // block b0
+  // }  // module m0
+  BlockBuilder builder;
+  const Operation& pair = builder.AddOperation(pair_op(), {}, {});
+  builder.AddOperation(plus_op(), {},
+                       {Value(value::OperationResult(pair, "first")),
+                        Value(value::OperationResult(pair, "second"))});
+  global_module().AddBlock(builder.build());
+  const auto& dot_graph_lines =
+      GetDotGraphAsWhitespaceTrimmedLines(global_module(), config);
+
+  EXPECT_THAT(
+      GetNodes(dot_graph_lines),
+      testing::UnorderedElementsAre(
+          // block b0
+          "B0",
+          // %0 = core.pair []()
+          R"(B0_0[label="{{}|core.pair|core.pair-2|{<first>first|<second>second}}"])",
+          // %1 = core.plus [](%0.first, %1.second)
+          R"(B0_1[label="{{<I0>I0|<I1>I1}|core.plus|core.plus-0|{<out>out}}"])"));
```

Comment:
If the result of an operation is a valueList, the usage of `%0.first` will be replaced with `%0`. Specifically, there will no longer be a value `ID.ID`. Is my understanding correct??

Reply:
That is correct.

Reply:
ok. Is the DotGeneratorConfig in this case provides additional label information of size of the operation result corresponding to the operator name? It was not obvious to me by reading the code. Does it make sense to add comments?

Reply:
Yes, it adds `core.plus-0` this text to the label. I will add a comment.

Reply:
Done, I also changed the format to `numResults(<op-name>): <num-results>`. It hopefully makes the additional annotation more obvious.